content/docs/gto/use-cases/model-registry.md

```diff
@@ -0,0 +1,62 @@
+# Machine Learning Model Registry
+
+A **model registry** is a tool to catalog ML models and their versions. Models
+from your data science projects can be discovered, tested, shared, deployed, and
+audited from there. [DVC](https://github.com/iterative/dvc), GTO, and [MLEM]
+enable these capabilities on top of Git, so you can stick to en existing
+software engineering stack. No more divide between ML engineering and
+operations!
+
+[mlem]: /doc
+
+ML model registries give your team key capabilities:
+
+- Collect and organize model [versions] from different sources effectively,
+  preserving their data provenance and lineage information.
+- Share metadata including [metrics and plots][mp] to help use and evaluate
+  models.
+- A standard interface to access all your ML artifacts, from early-stage
+  [experiments] to production-ready models.
+- Deploy specific models on different environments (dev, shadow, prod, etc.)
+  without touching the applications that consume them.
+- For security, control who can manage models, and audit their usage trails.
+
+[versions]: https://dvc.org/doc/use-cases/versioning-data-and-model-files
```

Comment:
Semantic versioning is the accepted way to version code. How should artifacts be versioned?
I have been asked this by a Data Scientist some time ago. Given that everyone is free to do whatever he wants, perhaps giving a hint is not bad...?

I formulated a reasonable convention for models, not sure if it could be of any use:

#### Patch
Model as a black-box is as before, it only outputs different numbers.

Typical scenario: model have been trained with more recent data
Typical scenario 2: changed hyper-parameters


#### Minor
May want to take advantage of additional outputs or additional functionalities

Typical scenario 1: model now has `predict_proba()` in addition to `predict()`
Typical scenario 2: model now outputs a json with an additional field `confidence_interval`, in addition to `predicted_values`


#### Major
Need to re-visit the code that calls the model to serve it (breaking change)

Typical scenario 1: model APIs have changed
Typical scenario 2: model expects different input data format
Typical scenario 3: model relies on different libraries, need to re-build the venv (or even the OS-level libraries)

Reply:
Good idea! We can make this advice a page in User Guide, I think. @jorgeorpinel, WDYT?

Reply:
It's a discussion for https://mlem.ai/doc/use-cases/model-registry I think. No need to repeat the use case page in here (it's already in the same site). Link to it from GTO docs as needed instead?

Reply:
Don't think it's for Use Cases - they're too high-level and these are details. IMHO

Reply:
Right. I meant I didn't think this file we're commenting on was needed here.

On ML SemVer, Idk. Seems opinionated to give such a precise recommendation. @francesco086 I encourage you to make a separate PR to contribute this in some existing or new page though, then the team can review it and decide.

It would probably belongs in GTO docs? That's what we use to annotate artifact versions right?

Reply:
I've created an issue to address this https://github.com/iterative/mlem.ai/issues/231

> That's what we use to annotate artifact versions right?

Yes